src/components/Admin/UserPage/ManageSensors.tsx

```diff
@@ -1,20 +1,208 @@
+      return unsubscribe;
+    }
+    return;
+  }, [isAuthenticated, isAdmin]);
+
+  /**
+   *
+   * @param event - click button event
+   * @param currentSensor - sensor for which to toggle isActive status
+   */
+  function toggleActiveSensorStatus(
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    currentSensor: Sensor
+  ) {
+    event.preventDefault();
+
+    if (isAdmin) {
+      firestore
+        .collection('sensors')
+        .doc(currentSensor.readingDocId)
+        .update({
+          isActive: !currentSensor.isActive,
+        })
+        .catch(() => {
+          setError(
+            t('sensors.changeActiveSensorError') + currentSensor.name ??
+              currentSensor.purpleAirId
+          );
+        });
+    }
+  }
+
+  /**
+   * Interface for ToggleActiveSensorPopover used for type safety
+   */
+  interface ToggleActiveSensorPopoverProps {
+    sensor: Sensor;
+  }
 
-  const {t} = useTranslation('administration');
+  /**
+   * Creates a button that when clicked, creates a confirmation popup to change
+   * a sensor's active status. Active means that data for the sensor will be
+   * collected and shown on the map, but does not change anything in PurpleAir.
+   * @param sensor - sensor for a row
+   */
+  const ToggleActiveSensorPopover: ({
+    sensor,
+  }: ToggleActiveSensorPopoverProps) => JSX.Element = ({
+    sensor,
+  }: ToggleActiveSensorPopoverProps) => {
+    const name = sensor.name ? sensor.name : sensor.purpleAirId;
+
+    const popoverMessage =
+      (sensor.isActive
+        ? t('sensors.confirmDeactivate')
+        : t('sensors.confirmActivate')) + name;
+    const popoverNote = sensor.isActive
+      ? t('sensors.deactivateNote')
+      : t('sensors.activateNote');
+
+    return (
+      <Popover>
+        <PopoverTrigger>
+          <Button colorScheme={sensor.isActive ? 'red' : 'green'} width="full">
+            {sensor.isActive ? t('sensors.deactivate') : t('sensors.activate')}
+          </Button>
+        </PopoverTrigger>
+        <PopoverContent>
+          <PopoverArrow />
+          <PopoverCloseButton />
+          <PopoverHeader>
+            <Heading fontSize="medium">{t('common:confirm')}</Heading>
+          </PopoverHeader>
+          <PopoverBody>
+            <Text>{popoverMessage}</Text>
+            <Divider marginY={1} />
+            <Text marginBottom={1}>{popoverNote}</Text>
+            <Center>
+              <Button
+                paddingY={2}
+                colorScheme={sensor.isActive ? 'red' : 'green'}
+                onClick={event => toggleActiveSensorStatus(event, sensor)}
+              >
+                {t('common:confirm')}
+              </Button>
+            </Center>
+          </PopoverBody>
+        </PopoverContent>
+      </Popover>
+    );
+  };
+
+  /**
+   *
+   * @param timestamp - the time to convert
+   * @returns human readable date time string, unknown if null
+   */
+  function timestampToDateString(
```

Comment:
I couldn't test this because there weren't any timestamps, but seems legit.

Reply:
Since I just merged the "add more data" PR, it now shows up!